Source/CSharpEssentials/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessCodeFixProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace CSharpEssentials.InvokeDelegateWithConditionalAccess
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(InvokeDelegateWithConditionalAccessCodeFixProvider)), Shared]
+    internal class InvokeDelegateWithConditionalAccessCodeFixProvider : CodeFixProvider
+    {
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DiagnosticIds.InvokeDelegateWithConditionalAccessId);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(CodeAction.Create(
+                "Delegate invocation can be simplified",
+                async c => await UpdateDocumentAsync(context).ConfigureAwait(false),
```

Comment:
I'm not sure this is quite right:

1. Why define an async lambda just to immediately await UpdateDocumentAsync? It seems reasonable you could just return UpdateDocumentAsync and let the code action engine do the waiting, no?
2. The lambda is passed a `CancellationToken` (`c`) but the code here chooses to use the `CancellationToken` provided by `context`. That might not work as you expect it to.